examples/example1/processors.py

```diff
@@ -1,9 +1,11 @@
 import datetime
 
+from lab_share_lib.processing.base_processor import BaseProcessor
 
-class Example1MessageProcessor:
-    def __init__(self, schema_registry, basic_publisher, config):
-        return
+
+class Example1MessageProcessor(BaseProcessor):
+    def instantiate(self, schema_registry, basic_publisher, config):
```

Comment:
This method is defined to be a static method by the parent BaseProcessor.

The _build_processor_for_subject method of RabbitMessageProcessor casts the processor to BaseProcessor. As the method is expected to be static, it will fail because it passes three arguments, which would not match the signature of Example1MessageProcessor's instantiate method. A quick workaround may be using the self argument from the method definition of Example1MessageProcessor's instantiate and maybe add @staticmethod decorator on it.

Otherwise, Example1MessageProcessor just returns an instance from the instantiate method and does not use the passed arguments. What is the point of passing schema_registry, basic_publisher and config arguments in this case?